py/src/test_call.py

```diff
@@ -622,6 +625,143 @@ def test_call_on_pending_deployed_through_existing():
+                "calldata": [132],
+                "gas_price": None,
+                "chain": StarknetChainId.MAINNET,
+                # because the block is not found, the updates are not used
+                "pending_updates": {
+                    contract_address: [
+                        {"key": 132, "value": 5},
+                    ]
+                },
+                "pending_deployed": maybe_pending_deployed([]),
+            },
+            [3],
+        )
+    )
+
+    # similar to above cases, but this time call on pending_deployed address.
+
+    commands.append(
+        (
+            {
+                "command": "call",
+                "at_block": existing_block,
+                "contract_address": 1234567,
+                "entry_point_selector": "get_value",
+                "calldata": [132],
+                "gas_price": None,
+                "chain": StarknetChainId.MAINNET,
+                "pending_updates": {
+                    1234567: [
+                        {"key": 132, "value": 5},
+                    ]
+                },
+                "pending_deployed": maybe_pending_deployed(
+                    [
+                        {
+                            "address": 1234567,
+                            "contract_hash": "0x050b2148c0d782914e0b12a1a32abe5e398930b7e914f82c65cb7afce0a0ab9b",
+                        }
+                    ]
+                ),
+            },
+            [5],
+        )
+    )
+
+    commands.append(
+        (
+            {
+                "command": "call",
+                # this block is not found
+                "at_block": reorgged_block,
+                "contract_address": 1234567,
+                "entry_point_selector": "get_value",
+                "calldata": [132],
+                "gas_price": None,
+                "chain": StarknetChainId.MAINNET,
+                # because the block is not found, the updates are not used
+                "pending_updates": {
+                    1234567: [
+                        {"key": 132, "value": 5},
+                    ]
+                },
+                "pending_deployed": maybe_pending_deployed(
+                    [
+                        {
+                            "address": 1234567,
+                            "contract_hash": "0x050b2148c0d782914e0b12a1a32abe5e398930b7e914f82c65cb7afce0a0ab9b",
+                        }
+                    ]
+                ),
+            },
+            NoSuchContract,
+        )
+    )
+
+    # existing test cases calling on non-existing blocks should work as they have been,
+    # because they don't define any value for pending stuffs
+
+    con.execute("BEGIN")
+    for command, expected in commands:
+        try:
+            (verb, output, _timings) = loop_inner(con, command)
+            assert expected == output
+        except NoSuchContract as e:
+            assert expected == NoSuchContract
```

Comment:
This is a bit of "rate-my-test-framework". I don't know if this could be done nicer. At least it isn't general purpose :)

Reply:
Fixed this. I think the comment is still valid but sadly not visible.